presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusClient.java

```diff
@@ -0,0 +1,185 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.prometheus;
+
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.io.Files;
+import io.airlift.json.JsonCodec;
+import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.connector.ColumnHandle;
+import io.prestosql.spi.predicate.TupleDomain;
+import io.prestosql.spi.type.DoubleType;
+import io.prestosql.spi.type.MapType;
+import io.prestosql.spi.type.TimestampType;
+import io.prestosql.spi.type.TypeManager;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+import org.apache.http.client.utils.URIBuilder;
+
+import javax.inject.Inject;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static io.prestosql.plugin.prometheus.PrometheusErrorCode.PROMETHEUS_TABLES_METRICS_RETRIEVE_ERROR;
+import static io.prestosql.plugin.prometheus.PrometheusErrorCode.PROMETHEUS_UNKNOWN_ERROR;
+import static io.prestosql.spi.StandardErrorCode.NOT_FOUND;
+import static io.prestosql.spi.type.TypeSignature.mapType;
+import static io.prestosql.spi.type.VarcharType.VARCHAR;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Objects.requireNonNull;
+
+public class PrometheusClient
+{
+    private static List<PrometheusColumn> prometheusColumns;
+    private static PrometheusConnectorConfig config;
+    static final String METRICS_ENDPOINT = "/api/v1/label/__name__/values";
+    private static final OkHttpClient httpClient = new OkHttpClient.Builder().build();
+    private final Supplier<Map<String, Object>> tableSupplier;
+    private Optional<TupleDomain<ColumnHandle>> pushedDown = Optional.empty();
+
+    @Inject
+    public PrometheusClient(PrometheusConnectorConfig config, JsonCodec<Map<String, Object>> metricCodec, TypeManager typeManager)
+            throws URISyntaxException
+    {
+        requireNonNull(config, "config is null");
+        requireNonNull(metricCodec, "metricCodec is null");
+        requireNonNull(typeManager, "typeManager is null");
+
+        tableSupplier = Suppliers.memoizeWithExpiration(metricsSupplier(metricCodec, getPrometheusMetricsURI(config)),
+                (long) config.getCacheDuration().getValue(), config.getCacheDuration().getUnit());
+        MapType varcharMapType = (MapType) typeManager.getType(mapType(VARCHAR.getTypeSignature(), VARCHAR.getTypeSignature()));
+        this.config = config;
+        this.prometheusColumns = ImmutableList.of(
```

Comment:
prometheusColumns shouldn't be initialized here since it's static.

Reply:
Fixed in dd3b9a84